src/storage/secondary/options.rs

```diff
@@ -83,7 +83,8 @@ impl StorageOptions {
             },
             checksum_type: ChecksumType::Crc32,
             encode_type: EncodeType::Plain,
-            record_first_key: false,
+            // required by range-filter scan rule
+            record_first_key: true,
```

Comment:
Looks like this feature is required by range-filter scan. Should we make it mandatory? or support range-filter scan without it? @skyzh